$OpenBSD: patch-src_TextClient_cpp,v 1.1 2017/08/11 11:23:57 dcoppa Exp $

commit eaa61e41dbfe4a83f29969d8d32b68b4c24f2e6e (tag: amule-svn-r10987)
Author: upstream svn <svn@amule.org>
Date:   Sun May 21 17:44:03 2017 +0000

Fix display of search results in amulecmd

Uses correct integer conversion identifiers in Printf

Index: src/TextClient.cpp
--- src/TextClient.cpp.orig
+++ src/TextClient.cpp
@@ -611,10 +611,10 @@ void CamulecmdApp::ShowResults(CResultMap results_map)
 		id = (*iter).first;
 		SearchFile* file = (*iter).second;
 
-		output.Printf(wxT("%i.      "), id);
+		output.Printf(wxT("%lu.      "), id);
 		output = output.SubString(0, nr_max).Append(file->sFileName).Append(' ', name_max);
-		mb.Printf(wxT("     %d"), file->lFileSize/1024/1024);
-		kb.Printf(wxT(".%d"), file->lFileSize/1024%1024);
+		mb.Printf(wxT("     %ld"), file->lFileSize/1024/1024);
+		kb.Printf(wxT(".%03ld"), file->lFileSize/1024%1024);
 		output = output.SubString(0, nr_max + name_max + mb_max - mb.Length() ).Append(mb).Append(kb);
 		printf("%s     %ld\n",(const char*)unicode2char(output), file->lSourceCount );
 	}
